spring-cloud-stream/src/main/java/org/springframework/cloud/stream/aggregate/AggregateApplicationUtils.java

```diff
@@ -59,10 +65,16 @@ static String getDefaultNamespace(String appClassName, int index) {
 	protected static SpringApplicationBuilder embedApp(
 			ConfigurableApplicationContext parentContext, String namespace,
 			Class<?> app) {
-		return new SpringApplicationBuilder(app).web(false).main(app).bannerMode(Mode.OFF)
+		// Child context needs to enable web MVC configuration and web enabled to obtain the MVC request mapping in the child applications
+		return new SpringApplicationBuilder(new Object[]{app, EmbeddedServletContainerAutoConfiguration.class,
+				EndpointWebMvcManagementContextConfiguration.class, WebMvcAutoConfiguration.EnableWebMvcConfiguration.class,
+				ServerPropertiesAutoConfiguration.class})
+				.web(true).main(app)
+				.bannerMode(Mode.OFF)
 				.properties("spring.jmx.default-domain=" + namespace)
 				.properties(
 						InternalPropertyNames.NAMESPACE_PROPERTY_NAME + "=" + namespace)
+				.properties("server.port=-1")
```

Comment:
In order to expose the web endpoints, the child applications need to be web `enabled` but here we make sure they don't expose the `server.port`. One possible way is to introduce a property to let the user decide whether to expose the web endpoints for the child application or not. That way, we will start web application context only for those child apps that have their web endpoints (request mappings + actuator endpoints) mapped to parent aggregate application.